conformance/lang/expressions/member-access-expr/xml_member_access_expr.balt

```diff
@@ -0,0 +1,1219 @@
+             container expression is an xml and evaluated value (k) of the key expression is greater than or equal to
+             its length (k >= length(c)).
+Labels: member-access-expr, value:toBalString, xml
+
+function init() {
+    xml seq = xml `<?xml-stylesheet type="text/xsl" href="style.xsl"?><name>Sherlock Holmes</name><!--Comment--><name>Enid Blyton</name>Some Text<name>Dan Brown</name>`;
+
+    // k == length(c)
+    io:println(seq[6].toBalString()); // @output xml``
+
+    // k > length(c)
+    io:println(seq[10].toBalString()); // @output xml``
+}
+
+Test-Case: output
+Description: Test the result of evaluating a member access expression when basic type of the evaluated value of the
+             container expression is a builtin subtype of xml.
+Labels: is-expr, member-access-expr, xml, xml:Comment, xml:Element, xml:length, xml:ProcessingInstruction, xml:Text
+
+function init() {
+    xml:ProcessingInstruction ins = xml `<?xml-stylesheet type="text/xsl" href="style.xsl"?>`;
+    io:println(ins[0] is xml); // @output true
+    io:println(ins[0].length()); // @output 1
+
+    xml:Element ele = xml `<name>Sherlock Holmes</name>`;
+    io:println(ele[0] is xml); // @output true
+    io:println(ele[0].length()); // @output 1
+
+    xml:Comment comment = xml `<!--Comment-->`;
+    io:println(comment[0] is xml); // @output true
+    io:println(comment[0].length()); // @output 1
+
+    xml:Text text = xml `Some Text`;
+    io:println(text[0] is xml); // @output true
+    io:println(text[0].length()); // @output 1
+}
+
+Test-Case: output
+Description: Test the result of evaluating a member access expression when basic type of the evaluated value (c) of the
+             container expression is a builtin subtype of xml and evaluated value (k) of the key expression is greater
+             than or equal to its length (k >= length(c)).
+Labels: member-access-expr, value:toBalString, xml:Comment, xml:Element, xml:ProcessingInstruction, xml:Text
+
+function init() {
+    xml:ProcessingInstruction ins = xml `<?xml-stylesheet type="text/xsl" href="style.xsl"?>`;
+    // k == length(c)
+    io:println(ins[1].toBalString()); // @output xml``
+    // k > length(c)
+    io:println(ins[4].toBalString()); // @output xml``
+
+    xml:Element ele = xml `<name>Sherlock Holmes</name>`;
+    // k == length(c)
+    io:println(ele[1].toBalString()); // @output xml``
+    // k > length(c)
+    io:println(ele[4].toBalString()); // @output xml``
+
+    xml:Comment comment = xml `<!--Comment-->`;
+    // k == length(c)
+    io:println(comment[1].toBalString()); // @output xml``
+    // k > length(c)
+    io:println(comment[4].toBalString()); // @output xml``
+
+    xml:Text text = xml `Some Text`;
+    // k == length(c)
+    io:println(text[1].toBalString()); // @output xml``
+    // k > length(c)
+    io:println(text[4].toBalString()); // @output xml``
+}
```

Comment:
Spec says,

> if it is xml, and k is < 0, then the evaluation completes abruptly with a panic; 

Do we have a test covering that? Seems in the implementation we do not panic for k<0, instead we return an empty XML value.